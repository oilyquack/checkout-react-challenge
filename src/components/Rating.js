import React from "react";

function Rating({ rating, ratingReceiver }) {
  function handleHover(event) {
    const hovered = event.target.id;

    for (let i = 1; i <= 5; i++) {
      const star = document.getElementById(i);
      if (i <= hovered) {
        star.innerText = "★";
      } else {
        star.innerText = "☆";
      }
    }
  }

  function handleLeave() {
    if (rating === null) {
      const stars = document.querySelectorAll(".star");

      stars.forEach(star => (star.innerText = "☆"));
    } else {
      for (let i = 1; i <= 5; i++) {
        const star = document.getElementById(i);
        if (i <= rating) {
          star.innerText = "★";
        } else {
          star.innerText = "☆";
        }
      }
    }
  }

  function handleClick(event) {
    const currentRating = event.target.id;
    ratingReceiver(currentRating);
  }

  return (
    <div id="app__form__rating">
      <span
        className="star"
        id="1"
        onMouseEnter={handleHover}
        onMouseLeave={handleLeave}
        onClick={handleClick}
      >
        ☆
      </span>
      <span
        className="star"
        id="2"
        onMouseEnter={handleHover}
        onMouseLeave={handleLeave}
        onClick={handleClick}
      >
        ☆
      </span>
      <span
        className="star"
        id="3"
        onMouseEnter={handleHover}
        onMouseLeave={handleLeave}
        onClick={handleClick}
      >
        ☆
      </span>
      <span
        className="star"
        id="4"
        onMouseEnter={handleHover}
        onMouseLeave={handleLeave}
        onClick={handleClick}
      >
        ☆
      </span>
      <span
        className="star"
        id="5"
        onMouseEnter={handleHover}
        onMouseLeave={handleLeave}
        onClick={handleClick}
      >
        ☆
      </span>
    </div>
  );
}

export default Rating;
