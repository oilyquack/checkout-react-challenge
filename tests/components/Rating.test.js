import React from "react";
import renderer from "react-test-renderer";
import Rating from "../../src/components/Rating";

it("matches the snapshot", () => {
  const tree = renderer.create(<Rating />);
  expect(tree).toMatchSnapshot();
});
