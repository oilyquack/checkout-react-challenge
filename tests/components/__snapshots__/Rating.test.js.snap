// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  id="app__form__rating"
>
  <span
    className="star"
    id="1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    ☆
  </span>
  <span
    className="star"
    id="2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    ☆
  </span>
  <span
    className="star"
    id="3"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    ☆
  </span>
  <span
    className="star"
    id="4"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    ☆
  </span>
  <span
    className="star"
    id="5"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    ☆
  </span>
</div>
`;